cmd/kube-proxy/app/server_test.go

```diff
@@ -374,3 +375,38 @@ func TestLoadConfigFailures(t *testing.T) {
 		}
 	}
 }
+
+// TestProcessHostnameOverrideFlag tests processing hostname-override arg
+func TestProcessHostnameOverrideFlag(t *testing.T) {
+
+	testCases := []struct {
+		name             string
+		hostnameArg      string
+		expectedHostname string
+	}{
+		{
+			name:             "Hostname from config file",
+			hostnameArg:      "",
+			expectedHostname: "foo",
+		},
+		{
+			name:             "Hostname from argument",
+			hostnameArg:      "  bar ",
+			expectedHostname: "bar",
+		},
+	}
+	for _, tc := range testCases {
+		options := NewOptions()
+		options.config = &kubeproxyconfig.KubeProxyConfiguration{
+			HostnameOverride: "foo",
+		}
+
+		options.hostNameOverrideArg = tc.hostnameArg
+
+		err := options.processHostnameOverrideFlag()
+		assert.NoError(t, err, "unexpected error for %s: %v", tc.name, err)
```

Comment:
When you switch to using a subtest, you can remove the initial `%s` for `tc.name`